packages/forms/src/widgets/MultiSelectTagWidget/MultiSelectTagWidget.js

```diff
@@ -1,106 +1,89 @@
 import PropTypes from 'prop-types';
 import React from 'react';
-import Badge from '@talend/react-components/lib/Badge';
-import { MenuItem } from 'react-bootstrap';
+import { Badge, Typeahead } from '@talend/react-components/lib';
 import classNames from 'classnames';
 import keycode from 'keycode';
 import theme from './MultiSelectTagWidget.scss';
 
-const ENTER = 'ENTER';
-const LEAVE = 'LEAVE';
+const DROP_DOWN_ITEM_HEIGHT = 49;
 
-const INPUT_TEXT_INDENT = 7.5;
-const INPUT_HEIGHT = 38;
-const INPUT_MIN_WIDTH = 135;
-const DROP_DOWN_ITEM_HEIGHT = 39;
-const DROP_DOWN_PADDING = 10;
+const escapeRegexCharacters = str => str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
 
-function mapValue2Label(enumOptions) {
-	return enumOptions.reduce((map, option) => {
-		map[option.value] = option.label; // eslint-disable-line no-param-reassign
-		return map;
-	}, {});
-}
-
-function DropDownOptions({
-	options, onSelectOption, filterText, createIfNoneMatch, onCreateNew, onMouseEvent,
-	selectedIndex, noAvailableMessage,
-}) {
-	const optionsToShow = options.map((item, index) => (
-		<MenuItem active={selectedIndex === index} key={index} onClick={() => onSelectOption(item)}>
-			{item.label}
-		</MenuItem>
-	));
-	const NO_AVAILABLE = (
-		<MenuItem onClick={() => onSelectOption()}>
-			{noAvailableMessage}
-		</MenuItem>
-	);
-	const NEW_OPTION = (
-		<MenuItem active onClick={onCreateNew}>
-			{filterText} (new)
-		</MenuItem>
-	);
+const mapValue2Label = enumOptions => enumOptions.reduce((map, option) => {
+	map[option.value] = option; // eslint-disable-line no-param-reassign
+	return map;
+}, {});
 
-	function renderOptions() {
-		if (optionsToShow.length > 0) {
-			return optionsToShow;
-		} else if (createIfNoneMatch && filterText.length > 0) {
-			return NEW_OPTION;
-		}
-		return NO_AVAILABLE;
+const getAllOptions = (options) => {
```

Comment:
please keep it as a named function, it help a lot since the function name is given if said function raise an error

Reply:
im not sure `getAllOptions` is a suitable name, since what this option does is returning a subset of the options Array or applying a transformation onto it brefore returning said subset.